### DISCUSSION TITLE
Compare columns in separate tables

### DISCUSSION THREAD

**Discussion by cmnord** [prisma/prisma#17551] [Q&A]
Hello,

The `fieldReference` preview feature lets you compare columns in the same table, not different tables. https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#compare-columns-in-the-same-table

It says:

> However, you can compare fields in separate models with [standard queries](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#model-queries).

How can I compare fields in separate models with standard queries, then?

For example, given the following schema of "listings" with associated "auctions" I would like to make a `findMany` query comparing an auctions column to a listings column.

SQL Schema (Prisma schema below):

```sql
CREATE TABLE listing (
    "id" TEXT NOT NULL,
    "price" INTEGER NOT NULL,
    CONSTRAINT "Listing_pkey" PRIMARY KEY ("id")
);

CREATE TABLE auction (
    "id" TEXT NOT NULL,
    "price" INTEGER NOT NULL,
    "listingid" TEXT NOT NULL,
    CONSTRAINT "Auction_pkey" PRIMARY KEY ("id")
);

ALTER TABLE auction ADD CONSTRAINT "Auction_listingId_fkey" FOREIGN KEY ("listingid") REFERENCES listing("id") ON DELETE CASCADE ON UPDATE CASCADE;

insert into listing (id, price) values (1, 100);
insert into listing (id, price) values (2, 200);

insert into auction (id, listingid, price) values (1, 1, 50);
insert into auction (id, listingid, price) values (2, 2, 200);
```

<details>

Prisma schema:

```prisma
model Listing {
  id                String    @id @default(cuid())
  price Int?

  auctions         Auction[]
}

model Auction {
  id String @id @default(cuid())
  price Int?

  listing   Listing @relation(fields: [listingId], references: [id], onDelete: Cascade, onUpdate: Cascade)
  listingId String
}
```

</details>

Desired SQL query ([Sqlfiddle](http://sqlfiddle.com/#!17/ccab7/1))

```sql
SELECT
    auction.id as auctionId,
    auction.listingId as listingId,
    auction.price as auctionPrice,
    listing.price as listingPrice
FROM auction LEFT JOIN listing on
auction.listingid = listing.id
WHERE
	auction.price / listing.price > 0.5; -- !!!! key where clause
```

I'm trying to do a similar query in the Prisma client, which looks something like:

```ts
await prisma.auction.findMany({
    where: {
        price: {
            gt: /* prisma.listing.fields.price * 0.5 */,
        },
    },
    include: {
        listing: true,
    },
});
```

What is the correct syntax to compare `prisma.auction.price` and its associated `prisma.listing.price`?

I am using Prisma version 4.7.1.

Thank you!

**Comment by cmnord** [comment]
Is this a duplicate of https://github.com/prisma/prisma/discussions/15637 ?

**Reply by nurul3101** [reply]
Yes, Both the discussions are similar, for now you would need to use [$queryRaw](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access) to compare columns that are in separate tables.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
